multigeiger/speaker.cpp

```diff
@@ -140,6 +142,7 @@ void IRAM_ATTR tick(bool high) {
   } else
     sequence = NULL;
 
+  portENTER_CRITICAL_ISR(&mux_audio);
   isr_tick_sequence = sequence;
   portEXIT_CRITICAL_ISR(&mux_audio);
```

Comment:
i thought about this:
- this was intended as an optimization (keep the mutex for a shorter time)
- sequence pointer is local, but it points to the global tick_sequence array.
- so i now think this change is incorrect: in-parallel modifications of the global array should be avoided by the mutex.
- maybe it is not actually causing a problem right now if we can assume that only one tick() call is running at a specific time, but i am not sure about how ISRs are processed on the ESP32.
- but from a structural pov it is better to have the mutex while mutating global stuff.

Reply:
this also needs checking in the other PR / in the new code that was added, maybe another similar problem there.